src/Core/AdminConsole/OrganizationAuth/Models/AuthRequestUpdateProcessor.cs

```diff
@@ -0,0 +1,106 @@
+﻿using System.ComponentModel.DataAnnotations;
+using System.Reflection;
+using Bit.Core.Auth.Models.Data;
+using Bit.Core.Enums;
+
+namespace Bit.Core.AdminConsole.OrganizationAuth.Models;
+
+public class AuthRequestUpdateProcessor
+{
+    public OrganizationAdminAuthRequest ProcessedAuthRequest { get; private set; }
+
+    private OrganizationAdminAuthRequest _unprocessedAuthRequest { get; }
+    private OrganizationAuthRequestUpdate _update { get; }
+    private AuthRequestUpdateProcessorConfiguration _configuration { get; }
+
+    public EventType OrganizationEventType => ProcessedAuthRequest?.Approved.Value ?? false
+        ? EventType.OrganizationUser_ApprovedAuthRequest
+        : EventType.OrganizationUser_RejectedAuthRequest;
+
+    public AuthRequestUpdateProcessor(
+        OrganizationAdminAuthRequest authRequest,
+        OrganizationAuthRequestUpdate update,
+        AuthRequestUpdateProcessorConfiguration configuration
+    )
+    {
+        _unprocessedAuthRequest = authRequest;
+        _update = update;
+        _configuration = configuration;
+    }
+
+    public void Process()
+    {
+        if (_unprocessedAuthRequest == null)
+        {
+            throw new AuthRequestUpdateCouldNotBeProcessedException();
+        }
+        var isExpired = DateTime.UtcNow >
+            _unprocessedAuthRequest.CreationDate
+            .Add(_configuration.AuthRequestExpiresAfter);
+        var isSpent = _unprocessedAuthRequest == null ||
```

Comment:
This null check is now unnecessary because it's already checked above.

Reply:
Implemented as suggested on https://github.com/bitwarden/server/pull/4064/commits/23b60d484640df0a44541b6ad268e83d1414e786